parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -418,27 +428,32 @@ def delete_used_properties(self):
         query = "MATCH (n) SET n.is_before = null, n.parents = null"
         self.session.run(query)
 
-    def create_fulltext_index(self):
-        query = """
-            CREATE FULLTEXT INDEX nodeSearchIds FOR (n:ENTRY) ON EACH [n.id]
-            OPTIONS {indexConfig: {`fulltext.analyzer`: 'keyword'}}
-        """
-        self.session.run(query)
+    def create_fulltext_index(self, multi_label, branch_name):
+        """Create indexes for search"""
+        index_name = multi_label.replace(":", "_")
+        query = [
+            f"""CREATE FULLTEXT INDEX {index_name+'_SearchIds'}
+            FOR (n:{'b_'+branch_name}) ON EACH [n.id]\n"""
+        ]
+        query.append("""OPTIONS {indexConfig: {`fulltext.analyzer`: 'keyword'}}""")
+        self.session.run("".join(query))
 
         language_codes = [lang.alpha2 for lang in list(iso639.languages) if lang.alpha2 != ""]
         tags_prefixed_lc = ["n.tags_" + lc + "_str" for lc in language_codes]
         tags_prefixed_lc = ", ".join(tags_prefixed_lc)
-        query = (
-            f"""CREATE FULLTEXT INDEX nodeSearchTags FOR (n:ENTRY) ON EACH [{tags_prefixed_lc}]"""
-        )
+        query = f"""CREATE FULLTEXT INDEX {index_name+'_SearchTags'}
+            FOR (n:{'b_'+branch_name}) ON EACH [{tags_prefixed_lc}]"""
         self.session.run(query)
 
-    def __call__(self, filename):
-        """process the file"""
-        self.create_nodes(filename)
-        self.create_child_link()
-        self.create_previous_link()
-        self.create_fulltext_index()
+    def __call__(self, filename, branch_name):
+        """Process the file"""
+        filename = self.normalized_filename(filename)
+        branch_name = self.normalizing(branch_name, char="_")
```

Comment:
It's pefectly ok to use "-" for branch name, indeed I think it's more readable.
I name my branches: taxonomy-feat-do-blah.

Reply:
Ah yeah, but neo4j doesn't allow hyphens while creating indexes. So, I changed to a underscore delimiter.